internal/pkg/template/templates/workloads/partials/cf/instancerole.yml

```diff
@@ -38,6 +38,27 @@ InstanceRole:
                 StringEquals:
                   'iam:ResourceTag/copilot-application': !Sub '${AppName}'
                   'iam:ResourceTag/copilot-environment': !Sub '${EnvName}'
+      {{- if hasSecrets .}}
+      - PolicyName: !Join ['', [!Ref AppName, '-', !Ref EnvName, '-', !Ref WorkloadName, SecretsPolicy]]
+        PolicyDocument:
+          Version: '2012-10-17'
+          Statement:
+            - Effect: 'Allow'
+              Action:
+                - 'ssm:GetParameters'
+              Resource:
+                - !Sub 'arn:${AWS::Partition}:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+            - Effect: 'Allow'
+              Action:
+                - 'secretsmanager:GetSecretValue'
+              Resource:
+                - !Sub 'arn:${AWS::Partition}:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*'
```

Comment:
I removed the condition for tags intentionally as we don't need these tags for secrets with App Runner service. Is it absolutely necessary to have them?

Reply:
I can add context in the mean time and Wanxian can provide more data if needed.

It's necessary to have the tags otherwise we're giving v broad permissions to app runner services created with Copilot. Today, Copilot ensures that users opt-in to reading secrets by forcing them to add `copilot-application` and `copilot-environment` tags to their secrets (see https://aws.github.io/copilot-cli/docs/developing/secrets/). To make the process easy, the `copilot secret init` command automatically does that.

Reply:
As discussed, I have added the Condition to check for tagged secrets.